client/src/types/home_viewer.ts

```diff
@@ -40,10 +48,21 @@ export const generateLeagueDataByAge = (
   | 'leaguePlayersGroupAge'
   | 'matchesByAge'
   | 'leagueTeams'
+  | 'leagueActive'
   | 'matches'
 > => {
   const leagueAgeKeys = map(leagues, (league) => league.leagueAgeType);
 
+  const leagueActive: { [key: string]: LeagueActiveHomeViewer } = {};
```

Comment:
this is not `leagueActive`.
All the league data passed from gql are `active` leagues so `leagueActive` doesn't look right.

And do you really need this in homeViewer? 
The nav bar can use 'leagueAgeKeys' <- this contains the age group available from the league data, and it can use that to list out league.

In `League.tsx` -> you will already have the league id so you can just grab the correct league from `homeViewer.leagues` after putting the raw leagues into homeViewer from gql result.

If you really need this rename it to `leagueActiveGroupByAge` or something related to what you are doing

Reply:
I created a `LeagueActive` because I needed league `id`, `name`, `age type`, `year` to make a nav bar and title, but as you said I just added a `homeViewer.leagues` to the `home_viewer.ts`.

Reply:
**If you really need this rename it to `leagueActiveGroupByAge` or something related to what you are doing**